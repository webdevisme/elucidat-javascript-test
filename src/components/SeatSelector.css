.SeatSelector {
    display: flex;
    position: relative;
    flex-wrap: wrap;
    list-style: none;
    padding-left: 0;
    padding-bottom: 50px;
    margin: 40px;
}
.SeatSelector li {
    width: 20%;
}
.SeatSelector::after {
    content: 'Screen';
    text-transform: uppercase;
    border-bottom: solid 4px #333;

    display: block;
    position: absolute;
    width: 100%;

    bottom: 0;
    text-align: center;
}
.legend {
    margin: 20px;
}
.legend span {
    margin-right: 20px;
    padding: 5px;
}
.legend .available {
    background-color: green;
}
.legend .taken {
    background-color: red;
}
