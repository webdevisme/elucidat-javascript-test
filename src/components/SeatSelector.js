import React, {
    useState
} from "react";
import './SeatSelector.css';
import Seat from './Seat.js';

function SeatSelector(props) {
    const [data] = useState({selectedSeat: null, seats: props.seats});
    
    return ( 
    <div>
        <div className="legend"><span className="available">Seat Available</span><span className="taken">Seat Taken</span></div>
        <ul className = "SeatSelector" >
            { data.seats.map(item => ( <li  key={ item.seatNumber }>
                <Seat data={item} ></Seat>
            </li>))
                }
        </ul >
    {data.selectedSeat ? <div className="Ticket"></div> : null}
    </div>
    );

}

export default SeatSelector;