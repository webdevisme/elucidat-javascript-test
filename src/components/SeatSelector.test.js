import React from 'react';
import renderer from 'react-test-renderer';
import SeatSelector from './SeatSelector';
import seatsData from '../../public/seats.json';

it('renders seats data passed to prop', () => {
  const component = renderer.create(<SeatSelector seats={seatsData}></SeatSelector>);
  
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
