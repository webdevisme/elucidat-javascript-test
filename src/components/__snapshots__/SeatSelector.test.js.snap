// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders seats data passed to prop 1`] = `
<div>
  <div
    className="legend"
  >
    <span
      className="available"
    >
      Seat Available
    </span>
    <span
      className="taken"
    >
      Seat Taken
    </span>
  </div>
  <ul
    className="SeatSelector"
  >
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          1A
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="false"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          2A
           : 
           Taken
        </div>
      </div>
    </li>
    <li>
      <div
        available="false"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          3A
           : 
           Taken
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          4A
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="false"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          5A
           : 
           Taken
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          1B
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="false"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          2B
           : 
           Taken
        </div>
      </div>
    </li>
    <li>
      <div
        available="false"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          3B
           : 
           Taken
        </div>
      </div>
    </li>
    <li>
      <div
        available="false"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          4B
           : 
           Taken
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          5B
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          1C
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          2C
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          3C
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          4C
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          5C
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          1D
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="false"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          2D
           : 
           Taken
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          3D
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          4D
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          5D
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          1E
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          2E
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="false"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          3E
           : 
           Taken
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          4E
           : 
          Available
        </div>
      </div>
    </li>
    <li>
      <div
        available="true"
        className="Seat"
        onClick={[Function]}
      >
        <img
          alt="Seat Icon"
          src="Seat.svg"
        />
        <div
          className="tooltip"
        >
          5E
           : 
          Available
        </div>
      </div>
    </li>
  </ul>
</div>
`;
